adaptor/mysql/reader.go

```diff
@@ -0,0 +1,233 @@
+	go func() {
+		defer close(out)
+		for {
+			select {
+			case c, ok := <-in:
+				if !ok {
+					return
+				}
+				log.With("db", db).With("table", c).Infoln("iterating...")
+				schemaTable := strings.Split(c, ".")
+				columnsResult, err := session.Query(fmt.Sprintf(`
+            SELECT COLUMN_NAME AS column_name, DATA_TYPE as data_type, "" as element_type
+            FROM INFORMATION_SCHEMA.COLUMNS
+            WHERE
+                TABLE_SCHEMA = '%v'
+            AND TABLE_NAME = '%v'
+            ORDER BY ORDINAL_POSITION;
+            `, schemaTable[0], schemaTable[1]))
+				// No element_types in mysql since no ARRAY data type
+				// at the moment we add an empty column to get the same layout as Postgres
+				// TODO: Update this code so we don't need that empty column?
+				// TODO: Use the driver to get column types? https://github.com/go-sql-driver/mysql#columntype-support
+				// No longer using that driver though
+				if err != nil {
+					log.With("db", db).With("table", c).Errorf("error getting columns %v", err)
+					continue
+				}
+				var columns [][]string
+				for columnsResult.Next() {
+					var columnName string
+					var columnType string
+					var columnArrayType sql.NullString // this value may be nil
+
+					err = columnsResult.Scan(&columnName, &columnType, &columnArrayType)
+					recoveredRegex := regexp.MustCompile("recovered")
+					if err != nil && !recoveredRegex.MatchString(err.Error()) {
+						log.With("table", c).Errorf("error scanning columns %v", err)
+						continue
+					}
+
+					column := []string{columnName, columnType}
+					columns = append(columns, column)
+					log.With("db", db).With("table", c).Debugln(columnName + ": " + columnType)
+				}
+
+				// build docs for table
+				docsResult, _ := session.Query(fmt.Sprintf("SELECT * FROM %v", c))
+
+				for docsResult.Next() {
+					dest := make([]interface{}, len(columns))
+					for i := range columns {
+						dest[i] = make([]byte, 30)
+						dest[i] = &dest[i]
+					}
+
+					var docMap map[string]interface{}
+					err = docsResult.Scan(dest...)
+					if err != nil {
+						log.With("table", c).Errorf("error scanning row %v", err)
+						continue
+					}
+
+					docMap = make(map[string]interface{})
+
+					for i, value := range dest {
+						xType := fmt.Sprintf("%T", value)
+						log.With("db", db).With("table", c).Debugf("value: %s, type: %s", value, xType)
+						switch value := value.(type) {
+						// Seems everything is []uint8
+						case []uint8:
+							docMap[columns[i][0]] = casifyValue(string(value), columns[i][1])
+						case string:
+							docMap[columns[i][0]] = casifyValue(string(value), columns[i][1])
+						default:
+							arrayRegexp := regexp.MustCompile("[[]]$")
+							if arrayRegexp.MatchString(columns[i][1]) {
+							} else {
+								docMap[columns[i][0]] = value
+							}
+						}
+					}
+					out <- doc{table: c, data: docMap}
+				}
+				log.With("db", db).With("table", c).Infoln("iterating complete")
```

Comment:
This log line seems to be causing an issue for Compose's internal use of Transporter. We might need to re-word it (even though it's the same as Postgresql and MongoDB).

Noting this down so it gets picked up in review.

Reply:
This has been resolved outside of this PR. I.e: issue was not with Transporter.